meilisearch-http/src/option.rs

```diff
@@ -175,6 +242,81 @@ impl Opt {
         !self.no_analytics
     }
 
+    /// Build a new Opt from config file, env vars and cli args.
+    pub fn build() -> Self {
+        // Parse the args to get the config_file_path.
+        let mut opts = Opt::parse();
+        if let Some(config_file_path) = opts.config_file_path.as_ref() {
+            eprintln!("loading config file : {:?}", config_file_path);
+            match std::fs::read(config_file_path) {
+                Ok(config) => {
+                    // If the arg is present, and the file successfully read, we deserialize it with `toml`.
+                    let opt_from_config =
+                        toml::from_slice::<Opt>(&config).expect("can't read file");
+                    // We inject the values from the toml in the corresponding env vars if needs be. Doing so, we respect the priority toml < env vars < cli args.
+                    opt_from_config.export_to_env();
+                    // Once injected we parse the cli args once again to take the new env vars into scope.
+                    opts = Opt::parse();
+                }
+                Err(err) => eprintln!("can't read {:?} : {}", config_file_path, err),
+            }
+        }
+
+        opts
+    }
+
+    /// Exports the opts values to their corresponding env vars if they are not set.
+    fn export_to_env(self) {
+        export_to_env_if_not_present(MEILI_DB_PATH, self.db_path);
```

Comment:
To make sure that we don't forget any struct field I would prefer that we destructure the `Opt` and list exactly *all of the fields*:

```rust
let Opt { db_path, env: _ } = self;
```